src/test/ui/lint/lint-ctypes-boxed-dst.rs

```diff
@@ -0,0 +1,11 @@
+#![deny(improper_ctypes_definitions)]
+
+pub extern "C" fn test_boxed_str() -> Box<str> { //~ ERROR
+    "test_boxed_str".into()
+}
+
+pub extern "C" fn test_boxed_slice() -> Box<[u8]> { //~ ERROR
+    b"test_boxed_slice"[..].into()
+}
+
```

Comment:
Maybe add a thin test for a thin pointer check here to show that it doesn't error?

Reply:
There are already tests for this in other files. I don't think duplicating one here is worthwhile?

Reply:
Alternatively I can merge the tests together in the already existing file.

Reply:
Added the tests in the already existing file for clarity.